### DISCUSSION TITLE
Plugin request method question

### DISCUSSION THREAD

**Discussion by kemeris2000** [lowdefy/lowdefy#1764] [Questions and Help] [answered]
Hi,

I am trying to make request from "lowdefy-plugin-jwt" plugin. Once I pass `methods: { request }` parameter to `authorize` function, plugin stops working. No error logs on server side, on client side I get:
`https://www.exmaple.com/api/auth/callback/jwt_provider 401 (Unauthorized)  `

```
import jwt from 'jsonwebtoken';
import createJwtToken from './createJwtToken.js';

function createAuthorize({ secret, issuer, roles, username, password, requestId }) {
  async function authorize({ token, username, password, methods: { request }, params: { requestId } }) {
    try {
      request(requestId);
      return "woks"
      };
    } catch (error) {
      console.log(error);
      throw error;
    }
  }
  return authorize;
}

export default createAuthorize;
```

however in my other plugin with custom Action everything is working fine:

```
function PollRequest({ methods: { request }, params: { requestId, delay = 3000, duration = 10000 } }) {
  var intervalId = setInterval(function(){
    request(requestId);
  }, delay);
  setTimeout(function(){
    clearInterval(intervalId);
  }, duration);
}

export default PollRequest;

```
How to properly call request from authorize function?
Thank you in advance.

**Comment by SamTolmay** [comment, marked as answer]
Hi @kemeris2000 

The Auth.js (next-auth) provider plugins don't receive methods like `request` as parameters.

The Provider only receives the properties defined in the `lowdefy.yaml`. This is done so that they are similar to other Auth.js examples.

What kind of request do you want to make? If it is an http/axios request you could use the Fetch api:
https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API/Using_Fetch

When using the credentials provider, the the `authorize` function receives variables passed by the `Login` action.

So to pass a username and password you would do:
```yaml
    - id: login
      type: Login
      params:
        providerId: jwt_provider
       username:
          _state: username
      password:
          _state: password
```

More on the credentials provider: https://next-auth.js.org/providers/credentials

**Reply by kemeris2000** [reply]
Thank you [SamTolmay](https://github.com/SamTolmay)
